invenio_app_ils/circulation/api.py

```diff
@@ -247,6 +250,30 @@ def checkout_loan(
     return pid, loan
 
 
+def bulk_extend_loans(patron_pid, **kwargs):
+    """Bulk extend qualified loans."""
+    loan_class = current_circulation.loan_record_cls
+    days = current_app.config["ILS_CIRCULATION_LOAN_WILL_EXPIRE_DAYS"]
+
+    loans_search = get_all_range_expiring_ils_loans_by_patron_pid(
+        expiring_in_days=days, patron_pid=patron_pid)
+
+    for loan in loans_search.scan():
+        loan_record = loan_class.get_record_by_pid(loan["pid"])
+        params = deepcopy(loan_record)
+        try:
+            extended_loan = current_circulation.circulation.trigger(
+                loan_record,
+                **dict(
+                    params,
+                    trigger="bulk_extend",
+                )
+            )
+            yield extended_loan
+        except Exception:
```

Comment:
if there is a problem with extension we simply ignore it, but maybe there are better ideas?

Reply:
I would return a list of success and failed so that it can be displayed to the user, otherwise the user might be tempted to click multiple times on the bulk extend button! :)